content/docs/security/ratelimit/global.md

```diff
@@ -0,0 +1,432 @@
+---
+title: Global rate limiting
+weight: 10
+description: Configure global rate limiting rules across all of your Gateways to protect the backing services in your cluster.
+---
+
+Configure global rate limiting rules across all of your Gateways to protect the backing services in your cluster.
+
+## About {#about}
+
+Global rate limiting in kgateway is powered by [Envoy's rate limiting service protocol](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rate_limit_filter). With global rate limiting, you can apply distributed, consistent rate limits across multiple Gateways. Unlike local rate limiting, which operates per Gateway instance, global rate limiting uses a central service to coordinate rate limits. Therefore, to use global rate limiting, you must bring your own rate limit service that implements the Envoy protocol.
+
+With your own rate limit service in place, you get benefits such as:
+
+- Coordinated rate limiting across multiple Gateways.
+- Centralized rate limit management with shared counters.
+- Dynamic descriptor-based rate limits that can consider multiple request attributes.
+- Consistent user experience, regardless of which Gateway receives the request.
+
+### Request flow
+
+Review the following sequence diagram to understand the request flow with global rate limiting.
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Gateway
+    participant RateLimiter
+    participant App
+
+    Client->>Gateway: 1. Send request to protected App
+    Gateway->>Gateway: 2. Receive request
+    Gateway->>RateLimiter: 3. Extract descriptors & send to Rate Limit Service
+    RateLimiter->>RateLimiter: 4. Apply configured limits for descriptors
+    RateLimiter->>Gateway: Return decision
+    alt Request allowed
+        Gateway->>App: 5. Forward request to App
+        App->>Gateway: Return response
+        Gateway->>Client: Return response to Client
+    else Rate limit reached
+        Gateway->>Client: 6. Deny request & return rate limit message
+    end
+```
+
+1. Client sends a request to an App that is protected by the Gateway.
+2. Gateway receives the request.
+3. Gateway extracts descriptors and sends them to the Rate Limit Service.
+4. The Rate Limit Service applies configured limits for those descriptors and returns a decision to the Gateway.
+5. If allowed, the Gateway forwards the request to the App.
+6. If the rate limit is reached, the Gateway denies the request and returns a message to the Client.
+
+### Architecture
+
+The global rate limiting feature consists of three components:
+
+1. **TrafficPolicy with rateLimit.global**: Configure your rate limit policy in a kgateway TrafficPolicy. The rate limit policy includes the descriptors to extract from requests for the Gateway to send to the Rate Limit Service.
+2. **GatewayExtension**: Configure a Gateway to use the Rate Limit Service by using a kgateway GatewayExtension.
```

Comment:
just checking if we need that in kgateway? cause in GG we seem to need it

Reply:
yes, it is needed in kgateway for things connecting the BYO rate limit service (or ext auth, ext proc)